### DISCUSSION TITLE
Upstream local DNS problem?

### DISCUSSION THREAD

**Discussion by Potterli20** [AdguardTeam/AdGuardHome#3518] [Q&A]
I used multiple ports locally, but only requested one port. I want to request other upstream ports.

How do I implement multiple ports, I use DNSProxy, only upstream one port, and then allocate several more ports
![Screenshot_20210830-010539_Chrome](https://user-images.githubusercontent.com/28734349/131259054-9ebe9cd8-42a1-4fe7-ac6c-611fac7ea680.png)
![Screenshot_20210830-010500_Chrome](https://user-images.githubusercontent.com/28734349/131259061-a6a80cb2-6e0e-4a6f-8f3d-89182c6639fb.png)

**Comment by ainar-g** [comment]
The upstream that you see in the  logs is the one the response from which the client actually got.  You can look at your `dnsproxy` logs, and you should see that they all get requested in parallel mode.

**Reply by Potterli20** [reply]
No, I don't have this diary on DNSProxy. I didn't open it. There is no diary on AdGuardHome.

**Reply by ainar-g** [reply]
If you made the same request, it was probably cached.  Try setting cache to `0` to see the requests actually being made.  But don't forget to enable cache again later.

**Reply by Potterli20** [reply]
Still no, I read the ADGH diary and only requested 127.0.0.1:5000

**Reply by Potterli20** [reply]
I also set the cache to zero

**Reply by Potterli20** [reply]
I set up multiple local ports in parallel, and it still doesn't work

**Reply by ainar-g** [reply]
We cannot reproduce this issue.  Please make sure that a `dnsproxy` runs on all of these ports and that all of these proxies show verbose logs (`dnsproxy -v`).

**Reply by Potterli20** [reply]
My dnsproxy is compiled by myself

**Reply by Potterli20** [reply]
Do you speak chinese? ? ?
I wrote some articles on bilibili

**Reply by ameshkov** [reply]
试试用中文写下这个问题的描述，我们会看看。情写**真的详细的**描述。

**Reply by Potterli20** [reply]
> 试试用中文写下这个问题的描述，我们会看看。情写**真的详细的**描述。

https://b23.tv/Y3rmdX
我是基于bind9搭建
然后adgh好像对本地dns端口进行高低解析

**Reply by Potterli20** [reply]
> 试试用中文写下这个问题的描述，我们会看看。情写**真的详细的**描述。

这个问题好迷惑，我从5000～5005和60～65的端口，adgh的日记优先是5000端口进行并行，没有看到其他端口，没有看到5001～5005和60～65的的端口

**Reply by Potterli20** [reply]
> 试试用中文写下这个问题的描述，我们会看看。情写**真的详细的**描述。

上游程序是dns proxy 上游dns采用dns文件
采用dns proxy的开多端口的特性，在adgh进行多端口并行处理

**Reply by Potterli20** [reply]
我设置多端口的原因是这个https://github.com/AdguardTeam/AdGuardHome/issues/3520

**Reply by ameshkov** [reply]
请在这儿上传 AGH的日志文件，来我们看看。

**Reply by Potterli20** [reply]
> 请在这儿上传 AGH的日志文件，来我们看看。

[AdGuardHome.txt](https://github.com/AdguardTeam/AdGuardHome/files/7091934/AdGuardHome.txt)

**Reply by Potterli20** [reply]
> 请在这儿上传 AGH的日志文件，来我们看看。

这个日记emmm，好像前几天清理了

**Reply by Potterli20** [reply]
> 请在这儿上传 AGH的日志文件，来我们看看。

直接给视频 https://t.me/adguard_en/141944

**Reply by ameshkov** [reply]
我们需要“verbose”水平的日志：https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#verboselog

**Reply by Potterli20** [reply]
> 我们需要“verbose”水平的日志：https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#verboselog

[AdGuardHome.txt](https://github.com/AdguardTeam/AdGuardHome/files/7097136/AdGuardHome.txt)

**Reply by ameshkov** [reply]
AGH初始化:
```
2021/09/02 15:46:21.604908 26185#906 [debug] Bootstraps: [9.9.9.10 149.112.112.10 2620:fe::10 2620:fe::fe:10]
2021/09/02 15:46:21.604954 26185#906 [debug] Upstream 0: 127.0.0.1:5000
2021/09/02 15:46:21.605002 26185#906 [debug] Upstream 1: 127.0.0.1:5353
2021/09/02 15:46:21.605045 26185#906 [debug] Upstream 2: 127.0.0.1:60
2021/09/02 15:46:21.605093 26185#906 [debug] Upstream 3: 127.0.0.1:61
2021/09/02 15:46:21.605139 26185#906 [debug] Upstream 4: 127.0.0.1:62
2021/09/02 15:46:21.605180 26185#906 [debug] Upstream 5: 127.0.0.1:63
```

上游服务器都是被用的：
```
2021/09/02 15:46:23.282888 26185#1380 [debug] 127.0.0.1:63: sending request A events-op-hicloud-com-drcn.appacc.dbankedge.net.
2021/09/02 15:46:23.285731 26185#1375 [debug] 127.0.0.1:5000: sending request A events-op-hicloud-com-drcn.appacc.dbankedge.net.
2021/09/02 15:46:23.286114 26185#1376 [debug] 127.0.0.1:5353: sending request A events-op-hicloud-com-drcn.appacc.dbankedge.net.
2021/09/02 15:46:23.286193 26185#1377 [debug] 127.0.0.1:60: sending request A events-op-hicloud-com-drcn.appacc.dbankedge.net.
2021/09/02 15:46:23.286262 26185#1378 [debug] 127.0.0.1:61: sending request A events-op-hicloud-com-drcn.appacc.dbankedge.net.
2021/09/02 15:46:23.286312 26185#1379 [debug] 127.0.0.1:62: sending request A events-op-hicloud-com-drcn.appacc.dbankedge.net.
```

**Reply by Potterli20** [reply]
> AGH初始化:
> 
> ```
> 2021/09/02 15:46:21.604908 26185#906 [debug] Bootstraps: [9.9.9.10 149.112.112.10 2620:fe::10 2620:fe::fe:10]
> 2021/09/02 15:46:21.604954 26185#906 [debug] Upstream 0: 127.0.0.1:5000
> 2021/09/02 15:46:21.605002 26185#906 [debug] Upstream 1: 127.0.0.1:5353
> 2021/09/02 15:46:21.605045 26185#906 [debug] Upstream 2: 127.0.0.1:60
> 2021/09/02 15:46:21.605093 26185#906 [debug] Upstream 3: 127.0.0.1:61
> 2021/09/02 15:46:21.605139 26185#906 [debug] Upstream 4: 127.0.0.1:62
> 2021/09/02 15:46:21.605180 26185#906 [debug] Upstream 5: 127.0.0.1:63
> ```
> 
> 上游服务器都是被用的：
> 
> ```
> 2021/09/02 15:46:23.282888 26185#1380 [debug] 127.0.0.1:63: sending request A events-op-hicloud-com-drcn.appacc.dbankedge.net.
> 2021/09/02 15:46:23.285731 26185#1375 [debug] 127.0.0.1:5000: sending request A events-op-hicloud-com-drcn.appacc.dbankedge.net.
> 2021/09/02 15:46:23.286114 26185#1376 [debug] 127.0.0.1:5353: sending request A events-op-hicloud-com-drcn.appacc.dbankedge.net.
> 2021/09/02 15:46:23.286193 26185#1377 [debug] 127.0.0.1:60: sending request A events-op-hicloud-com-drcn.appacc.dbankedge.net.
> 2021/09/02 15:46:23.286262 26185#1378 [debug] 127.0.0.1:61: sending request A events-op-hicloud-com-drcn.appacc.dbankedge.net.
> 2021/09/02 15:46:23.286312 26185#1379 [debug] 127.0.0.1:62: sending request A events-op-hicloud-com-drcn.appacc.dbankedge.net.
> ```

但我的adgh的日记里都5000的端口请求，这很奇怪也。视频里也看了里，我不管怎么上游，都是优先5000，其他端口是正常的

**Reply by ameshkov** [reply]
在并行请求模式，Web的请求日记只表示AGH用回应的请求，别的请求你只能看在日志文件里。

**Reply by ameshkov** [reply]
> 都是优先5000

有可能5000的上游服务器简单更快。

**Reply by Potterli20** [reply]
> > 都是优先5000
> 
> 有可能5000的上游服务器简单更快。

5000和其他端口一样的，我只单用5000的话，有一些网站没有处理过来，所以不喜欢用单用一个端口，想用多端并行处理

**Reply by ameshkov** [reply]
你的日志文件我没看任何问题，AGH用不同的上游服务器。

例子：
```
2021/09/02 15:46:23.344017 26185#1410 [debug] 127.0.0.1:63: sending request A smpimg-1258344701.file.myqcloud.com.cdn.dnsv1.com.
2021/09/02 15:46:23.344101 26185#1405 [debug] 127.0.0.1:5000: sending request A smpimg-1258344701.file.myqcloud.com.cdn.dnsv1.com.
2021/09/02 15:46:23.344255 26185#1406 [debug] 127.0.0.1:5353: sending request A smpimg-1258344701.file.myqcloud.com.cdn.dnsv1.com.
2021/09/02 15:46:23.344355 26185#1407 [debug] 127.0.0.1:60: sending request A smpimg-1258344701.file.myqcloud.com.cdn.dnsv1.com.
2021/09/02 15:46:23.344425 26185#1408 [debug] 127.0.0.1:61: sending request A smpimg-1258344701.file.myqcloud.com.cdn.dnsv1.com.
2021/09/02 15:46:23.344491 26185#1409 [debug] 127.0.0.1:62: sending request A smpimg-1258344701.file.myqcloud.com.cdn.dnsv1.com.
2021/09/02 15:46:23.344664 26185#1394 [debug] 127.0.0.1:61: response: ok
```

`127.0.0.1:61`是用的。

```
2021/09/02 15:46:23.357091 26185#1417 [debug] 127.0.0.1:63: sending request A psixmlzn.sched.sma.tdnsv5.com.
2021/09/02 15:46:23.357176 26185#1412 [debug] 127.0.0.1:5000: sending request A psixmlzn.sched.sma.tdnsv5.com.
2021/09/02 15:46:23.357313 26185#1413 [debug] 127.0.0.1:5353: sending request A psixmlzn.sched.sma.tdnsv5.com.
2021/09/02 15:46:23.357375 26185#1414 [debug] 127.0.0.1:60: sending request A psixmlzn.sched.sma.tdnsv5.com.
2021/09/02 15:46:23.357452 26185#1415 [debug] 127.0.0.1:61: sending request A psixmlzn.sched.sma.tdnsv5.com.
2021/09/02 15:46:23.357497 26185#1416 [debug] 127.0.0.1:62: sending request A psixmlzn.sched.sma.tdnsv5.com.
2021/09/02 15:46:23.357758 26185#1407 [debug] 127.0.0.1:60: response: ok
```

`127.0.0.1:60`是用的。

等等等。

**Reply by Potterli20** [reply]
> 你的日志文件我没看任何问题，AGH用不同的上游服务器。
> 
> 例子：
> 
> ```
> 2021/09/02 15:46:23.344017 26185#1410 [debug] 127.0.0.1:63: sending request A smpimg-1258344701.file.myqcloud.com.cdn.dnsv1.com.
> 2021/09/02 15:46:23.344101 26185#1405 [debug] 127.0.0.1:5000: sending request A smpimg-1258344701.file.myqcloud.com.cdn.dnsv1.com.
> 2021/09/02 15:46:23.344255 26185#1406 [debug] 127.0.0.1:5353: sending request A smpimg-1258344701.file.myqcloud.com.cdn.dnsv1.com.
> 2021/09/02 15:46:23.344355 26185#1407 [debug] 127.0.0.1:60: sending request A smpimg-1258344701.file.myqcloud.com.cdn.dnsv1.com.
> 2021/09/02 15:46:23.344425 26185#1408 [debug] 127.0.0.1:61: sending request A smpimg-1258344701.file.myqcloud.com.cdn.dnsv1.com.
> 2021/09/02 15:46:23.344491 26185#1409 [debug] 127.0.0.1:62: sending request A smpimg-1258344701.file.myqcloud.com.cdn.dnsv1.com.
> 2021/09/02 15:46:23.344664 26185#1394 [debug] 127.0.0.1:61: response: ok
> ```
> 
> `127.0.0.1:61`是用的。
> 
> ```
> 2021/09/02 15:46:23.357091 26185#1417 [debug] 127.0.0.1:63: sending request A psixmlzn.sched.sma.tdnsv5.com.
> 2021/09/02 15:46:23.357176 26185#1412 [debug] 127.0.0.1:5000: sending request A psixmlzn.sched.sma.tdnsv5.com.
> 2021/09/02 15:46:23.357313 26185#1413 [debug] 127.0.0.1:5353: sending request A psixmlzn.sched.sma.tdnsv5.com.
> 2021/09/02 15:46:23.357375 26185#1414 [debug] 127.0.0.1:60: sending request A psixmlzn.sched.sma.tdnsv5.com.
> 2021/09/02 15:46:23.357452 26185#1415 [debug] 127.0.0.1:61: sending request A psixmlzn.sched.sma.tdnsv5.com.
> 2021/09/02 15:46:23.357497 26185#1416 [debug] 127.0.0.1:62: sending request A psixmlzn.sched.sma.tdnsv5.com.
> 2021/09/02 15:46:23.357758 26185#1407 [debug] 127.0.0.1:60: response: ok
> ```
> 
> `127.0.0.1:60`是用的。
> 
> 等等等。

我的5000好像是用dnsmasq来当服务器，然后用dnsproxy当开多端口服务器：例  dnsproxy端口有60.61.62.63.64等，只上游dnsmasq的5000，最后到adgh上把5000.60.61.62.63.64等多个端口设置为上游服务器。但在adgh的面板上只日记上只有5000的端口日记，所以只认为5000只有效果其他就是没有效果。所以我更想有多个端口上游请求

**Reply by Potterli20** [reply]
同时我也实验三种处理方法，都是只有5000为上游请求，其他端口都没有看到日记

**Reply by Potterli20** [reply]
有了多个端口处理请求，网站快了不少

**Reply by ameshkov** [reply]
> 但在adgh的面板上只日记上只有5000的端口日记

眼睛不可靠，如果你要真的了解情况，查看日志文件。

**Reply by Potterli20** [reply]
> > 但在adgh的面板上只日记上只有5000的端口日记
> 
> 眼睛不可靠，如果你要真的了解情况，查看日志文件。

哈哈哈哈，那好叭😂
那个dnsproxy反代理不行哈，我崩溃了，证书不生效
